pykube/http.py

```diff
@@ -53,7 +59,13 @@ def build_session(self):
         elif "insecure-skip-tls-verify" in self.config.cluster:
             s.verify = not self.config.cluster["insecure-skip-tls-verify"]
         if "token" in self.config.user and self.config.user["token"]:
-            s.headers["Authorization"] = "Bearer {}".format(self.config.user["token"])
+            self._set_bearer_token(s, self.config.user["token"])
+        elif "auth-provider" in self.config.user:
+            try:
+                token = self.config.user['auth-provider']['config']['access-token']
+                self._set_bearer_token(s, token)
+            except KeyError:
```

Comment:
I am not a fan of this exception style. Could you adjust this to `token = self.config["auth-provider"].get("config", {}).get("access-token")` then setting the bearer token when token is not None. This IMO is cleaner and more explicit.